### ISSUE TITLE
Weird tracing results

### ISSUE THREAD

**Issue by voroninp** [dotnet/aspnetcore#48915]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I use gRPC service with a server streaming endpoint. And the client's span duration is much longer than the server's one while the only thing client does is reading all items from the stream.

![image](https://github.com/dotnet/aspnetcore/assets/9202028/2dd7ae90-c2da-41ba-b204-04d05989e808)

It looks like ASP.NET Core instrumentation does not take stream lifetime into account, whereas client gRPC instrumentation does.

### Expected Behavior

I expect span to live as long as gRPC stream does.

**Comment by JamesNK** [comment]
gRPC doesn't create its own distributed tracing. It uses the default activity that ASP.NET Core hosting creates.

As far as I know, the ASP.NET Core activity is stopped when the response is finished, which is the entire length of the streaming response.

Are you sure you have correctly identified each span?

**Comment by voroninp** [comment]
@JamesNK  Yes. There is an adjacent span on the client side which has the same duration as client's gRPC call.
